compiler/rustc_mir/src/transform/check_consts/validation.rs

```diff
@@ -876,15 +871,6 @@ impl Visitor<'tcx> for Validator<'mir, 'tcx> {
 
                 let is_intrinsic = tcx.fn_sig(callee).abi() == RustIntrinsic;
 
-                // HACK: This is to "unstabilize" the `transmute` intrinsic
-                // within const fns. `transmute` is allowed in all other const contexts.
-                // This won't really scale to more intrinsics or functions. Let's allow const
-                // transmutes in const fn before we add more hacks to this.
-                if is_intrinsic && tcx.item_name(callee) == sym::transmute {
-                    self.check_op(ops::Transmute);
```

Comment:
> I tried to make the lint forbid-by-default, but the compiler complained.

Oh right, making it a hard error is not a breaking change, nice!

I think the right place would be right here. Check the generic parameters for the patterns we want to forbid and report a non-const-op. It is after all a thing that is only relevant in const contexts

Reply:
Sounds reasonable. Wording will likely be a bit odd I'd imagine. Should it remain behind a feature flag or just be outright forbidden?

Reply:
Hm, I am not sure I like putting this here. This file is doing a static analysis that ensures certain properties. The lint does not actually ensure anything, it's just a best-effort thing with lots of false negatives.